packages/date-picker/test/validation.common.js

```diff
@@ -347,9 +353,77 @@ describe('validation', () => {
       expect(datePicker.invalid).to.be.true;
 
       // Try valid value.
-      datePicker.value = '2016-01-01';
+      datePicker.value = '2016-02-01';
       expect(datePicker.validate()).to.be.true;
       expect(datePicker.invalid).to.be.false;
     });
   });
+
+  describe('keyboard navigation', () => {
+    let focusSpy;
+
+    beforeEach(async () => {
+      datePicker = fixtureSync(`<vaadin-date-picker min="2010-01-01" max="2010-01-31"></vaadin-date-picker>`);
+      await nextRender();
+      await open(datePicker);
+      focusSpy = sinon.spy(datePicker._overlayContent, 'focusDate');
+      datePicker.isDateDisabled = (date) => {
+        if (!date) {
+          return false;
+        }
+        return date.year === 2010 && date.month === 0 && date.day === 29;
+      };
+    });
+
+    it('should not allow navigation prior to min', async () => {
```

Comment:
These first two tests just codify the existing keyboard navigation below min or above max.